src/telemetry.c

```diff
@@ -0,0 +1,61 @@
+
+/*
+ *   Software Updater - telemetry collection
+ *
+ *      Copyright © 2012-2016 Intel Corporation.
+ *
+ *   This program is free software: you can redistribute it and/or modify
+ *   it under the terms of the GNU General Public License as published by
+ *   the Free Software Foundation, version 2 or later of the License.
+ *
+ *   This program is distributed in the hope that it will be useful,
+ *   but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *   GNU General Public License for more details.
+ *
+ *   You should have received a copy of the GNU General Public License
+ *   along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ *   Authors:
+ *         Auke Kok <auke-jan.h.kok@intel.com>
+ *
+ */
+
+#define _GNU_SOURCE
+#include <stdlib.h>
+#include <unistd.h>
+#include <stdarg.h>
+
+#include "config.h"
+#include "swupd.h"
+
+#define RECORD_VERSION 1
+
+void telemetry(telem_prio_t level, const char *fmt, ...)
+{
+	va_list args;
+	char *filename;
+	int fd;
+
+	string_or_die(&filename, "%s/telemetry/%d.%d.XXXXXX", state_dir, RECORD_VERSION, level);
+
+	if (!filename) {
```

Comment:
`filename` must be non-null at this point, because `string_or_die` aborts if memory allocation fails.